src/mscorlib/shared/System/Text/StringBuilder.cs

```diff
@@ -213,16 +220,13 @@ private StringBuilder(SerializationInfo info, StreamingContext context)
                         capacityPresent = true;
                         break;
                     default:
-                        // Ignore other fields for forward compatibility.
+                        // Ignore other fields for forwards-compatibility.
                         break;
                 }
             }
 
             // Check values and set defaults
-            if (persistedString == null)
-            {
-                persistedString = String.Empty;
-            }
+            persistedString = persistedString ?? string.Empty;
```

Comment:
Same comment about unnecessary writes.  Yes, this is slightly shorter C#, but at the expense of more work (even if small).